### ISSUE TITLE
(core): bug with BundlingDockerImage.cp

### ISSUE THREAD

**Issue by bigkraig** [aws/aws-cdk#13273]
BundlingDockerImage.cp uses dockerExec to [create a container](https://github.com/aws/aws-cdk/blob/5d71d8e815529ccde7ffe13bb72f57ae88e896dc/packages/%40aws-cdk/core/lib/bundling.ts#L203) from the image to cp files out of. dockerExec was modified in https://github.com/aws/aws-cdk/pull/12001 to show the stdout/stderr, which has prevented the stdout from being returned.

I've submitted a PR that fixes the issue https://github.com/aws/aws-cdk/pull/13271

### Reproduction Steps

Sample repo to reproduce the issue:
https://github.com/bigkraig/cdk-docker-cp

### What did you expect to happen?

cp to copy the file from the image.

### What actually happened?

The container id is printed, but not returned to cp to continue.
```
$ cdk synth
[+] Building 1.2s (7/7) FINISHED                                                                                                                                                           
 => [internal] load build definition from Dockerfile                                                                                                                                  0.0s
 => => transferring dockerfile: 36B                                                                                                                                                   0.0s
 => [internal] load .dockerignore                                                                                                                                                     0.0s
 => => transferring context: 2B                                                                                                                                                       0.0s
 => [internal] load metadata for docker.io/library/alpine:latest                                                                                                                      1.1s
 => [auth] library/alpine:pull token for registry-1.docker.io                                                                                                                         0.0s
 => [1/2] FROM docker.io/library/alpine@sha256:a75afd8b57e7f34e4dad8d65e2c7ba2e1975c795ce1ee22fa34f8cf46f96a3be                                                                       0.0s
 => CACHED [2/2] RUN echo test > /hello.txt                                                                                                                                           0.0s
 => exporting to image                                                                                                                                                                0.0s
 => => exporting layers                                                                                                                                                               0.0s
 => => writing image sha256:7b67c7b2cffd9a5181e1561ba427333e42333dd5b68e180e20524593bed83450                                                                                          0.0s
 => => naming to docker.io/library/cdk-ef58d8b98ca16cdb3a02a7184de50cd6f36ed0e8d69b061d0d84436cbe909939                                                                               0.0s
b57ec5676652d885c89e2a23bae3113085b2cb595fe4737f0b3ce18ef9866894
TypeError: Cannot read property 'toString' of null
    at BundlingDockerImage.cp (/Users/kamador/src/cdk-docker-cp/node_modules/@aws-cdk/core/lib/bundling.ts:201:26)
    at new CdkDockerCpStack (/Users/kamador/src/cdk-docker-cp/lib/cdk-docker-cp-stack.ts:10:11)
    at Object.<anonymous> (/Users/kamador/src/cdk-docker-cp/bin/cdk-docker-cp.ts:7:1)
    at Module._compile (node:internal/modules/cjs/loader:1091:14)
    at Module.m._compile (/Users/kamador/src/cdk-docker-cp/node_modules/ts-node/src/index.ts:1056:23)
    at Module._extensions..js (node:internal/modules/cjs/loader:1120:10)
    at Object.require.extensions.<computed> [as .ts] (/Users/kamador/src/cdk-docker-cp/node_modules/ts-node/src/index.ts:1059:12)
    at Module.load (node:internal/modules/cjs/loader:971:32)
    at Function.Module._load (node:internal/modules/cjs/loader:812:14)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:76:12)
npm ERR! code 1
npm ERR! path /Users/kamador/src/cdk-docker-cp
npm ERR! command failed
npm ERR! command sh -c ts-node "--prefer-ts-exts" "bin/cdk-docker-cp.ts"

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/kamador/.npm/_logs/2021-02-24T23_59_07_006Z-debug.log
Subprocess exited with error 1
```

### Environment

  - **CDK CLI Version  :** 1.91.0
  - **Framework Version:** 1.91.0
  - **Node.js Version:** v15.9.0
  - **OS               :** MacOS and in CodeFresh CICD
  - **Language (Version):** TypeScript 4.1.5

### Other

--- 

This is :bug: Bug Report

**Comment by bigkraig** [comment]
Will be corrected by #13152

**Comment by jogold** [comment]
Should be fixed once #12258 (reverted) is redone.
